shootout/vampir-plonk/src/lib.rs

```diff
@@ -0,0 +1,107 @@
+use plonk_core::prelude::VerifierData;
+use plonk_core::proof_system::{ProverKey, VerifierKey, Proof};
+use plonk_core::proof_system::pi::PublicInputs;
+use plonk_core::error::Error;
+use plonk::error::to_pc_error;
+use ark_bls12_381::{Bls12_381, Fr as BlsScalar};
+use ark_ed_on_bls12_381::EdwardsParameters as JubJubParameters;
+use ark_poly_commit::{kzg10::UniversalParams, sonic_pc::SonicKZG10, PolynomialCommitment};
+use ark_poly::polynomial::univariate::DensePolynomial;
+use plonk_core::circuit::{Circuit, verify_proof};
+
+use std::collections::HashMap;
+use std::fs;
+use std::rc::Rc;
+use rand_core::OsRng;
+
+use vamp_ir::plonk::synth::PlonkModule;
+use vamp_ir::ast::Module;
+use vamp_ir::transform::compile;
+use vamp_ir::plonk::synth::{PrimeFieldOps, make_constant};
+use vamp_ir::util::prompt_inputs;
+
+type PC = SonicKZG10<Bls12_381, DensePolynomial<BlsScalar>>;
+
+static NAME: &'static str = "Vamp-IR zk-garage plonk: Blake2s";
+
+#[derive(Clone)]
+pub struct Blake2sCircuit<'a> {
+    pub path: &'a str,
+}
```

Comment:
turn these `&str` into `String` See the other backends as they handle locations fine with `String`